node-bootstrapper/use_kubeconfig.go

```diff
@@ -0,0 +1,396 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/Azure/agentbaker/pkg/agent"
+	"github.com/Azure/agentbaker/pkg/agent/datamodel"
+	yaml "sigs.k8s.io/yaml/goyaml.v3"
+)
+
+func useKubeconfig(config *datamodel.NodeBootstrappingConfiguration, files map[string]File) error {
+	if err := useKubeConfig(config, files); err != nil {
+		return err
+	}
+	switch config.BootstrappingMethod {
+	case datamodel.UseArcMsiToMakeCSR,
+		datamodel.UseArcMsiDirectly:
+		if err := useArcTokenSh(config, files); err != nil {
+			return err
+		}
+
+	case datamodel.UseAzureMsiDirectly,
+		datamodel.UseAzureMsiToMakeCSR:
+		if err := useAzureTokenSh(config, files); err != nil {
+			return err
+		}
+
+	case datamodel.UseTLSBootstrapToken, datamodel.UseSecureTLSBootstrapping:
+		break
+
+	default:
+		break
```

Comment:
do we want to make this field mandatory in the new contract?

Reply:
We probably should. Yes.